src/EnergyPlus/UnitarySystem.cc

```diff
@@ -2883,4287 +2898,4232 @@ namespace UnitarySystems {
+                }
+                if (fields.find("fraction_of_on_cycle_power_use") != fields.end()) { // not required field, has default
+                    input_spec.fraction_of_on_cycle_power_use = fields.at("fraction_of_on_cycle_power_use");
+                }
+                if (fields.find("heat_pump_fan_delay_time") != fields.end()) { // not required field, has default
+                    input_spec.heat_pump_fan_delay_time = fields.at("heat_pump_fan_delay_time");
+                }
+                if (fields.find("ancillary_on_cycle_electric_power") != fields.end()) { // not required field, has default
+                    input_spec.ancillary_on_cycle_electric_power = fields.at("ancillary_on_cycle_electric_power");
+                }
+                if (fields.find("ancillary_off_cycle_electric_power") != fields.end()) { // not required field, has default
+                    input_spec.ancillary_off_cycle_electric_power = fields.at("ancillary_off_cycle_electric_power");
+                }
+                if (fields.find("design_heat_recovery_water_flow_rate") != fields.end()) { // not required field, has default
+                    input_spec.design_heat_recovery_water_flow_rate = fields.at("design_heat_recovery_water_flow_rate");
+                }
+                if (fields.find("maximum_temperature_for_heat_recovery") != fields.end()) { // not required field, has default
+                    input_spec.maximum_temperature_for_heat_recovery = fields.at("maximum_temperature_for_heat_recovery");
+                }
+                if (fields.find("heat_recovery_water_inlet_node_name") != fields.end()) { // not required field
+                    input_spec.heat_recovery_water_inlet_node_name = UtilityRoutines::MakeUPPERCase(fields.at("heat_recovery_water_inlet_node_name"));
+                }
+                if (fields.find("heat_recovery_water_outlet_node_name") != fields.end()) { // not required field
+                    input_spec.heat_recovery_water_outlet_node_name =
+                        UtilityRoutines::MakeUPPERCase(fields.at("heat_recovery_water_outlet_node_name"));
+                }
+                if (fields.find("design_specification_multispeed_object_type") != fields.end()) { // not required field
+                    input_spec.design_specification_multispeed_object_type =
+                        UtilityRoutines::MakeUPPERCase(fields.at("design_specification_multispeed_object_type"));
+                }
+                if (fields.find("design_specification_multispeed_object_name") != fields.end()) { // not required field
+                    input_spec.design_specification_multispeed_object_name =
+                        UtilityRoutines::MakeUPPERCase(fields.at("design_specification_multispeed_object_name"));
+                }
+
+                thisSys.processInputSpec(state, input_spec, sysNum, errorsFound, ZoneEquipment, ZoneOAUnitNum);        
```

Comment:
Moved input processing to processInputSpec. This should be able to be called by alternate getInput functions.